### PR TITLE
#309: returning the forms patterns

### PR DESCRIPTION
**Pull request by hani-ibrahim**
working on #309

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/AddressFormType.swift [hunk 1]
```diff
@@ -132,7 +132,7 @@ enum AddressFormField: String {
         case .City:
             return [.Required,
                     .MaxLength(maxLength: 50),
-                    .Pattern(pattern: "^[,;()'0-9 \\w-]+$")]
+                    .Pattern(pattern: "^["+FormValidator.anyCharacterPattern+"]'?[-,;()' 0-9"+FormValidator.anyCharacterPattern+"ß]+$")]
```
Comment (bonkey) [inline AtlasUI/AtlasUI/AddressFormType.swift]: All those three are same, right? So at least don't DRY and create

```
enum FormValidator {
  internal static let anyNamePattern = "^[\(FormValidator.anyCharacterPattern)]'?[- \(FormValidator.anyCharacterPattern)ß]+$"
}
```

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/AddressFormType.swift (hunk 1 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AddressFormType.swift]: only the first two are the same and the third is different

Comment (bonkey) [inline AtlasUI/AtlasUI/AddressFormType.swift]: I'm not trying to be pain-in-the-butt, but it just proves it should be extracted to two separated and named patterns: name, city), as it's pretty unreadable this way and I didn't spot the difference.

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AddressFormType.swift]: completely agree

**Comment by hani-ibrahim** [comment]
👍

**Comment by bonkey** [comment]
👍

**Pull request merged**